lib/cli/src/Cardano/CLI.hs

```diff
@@ -395,189 +337,428 @@ runCliCommand Environment {..} manager execServe execLaunch
 -- | Execute 'launch' commands. This differs from the 'serve' command as it
 -- takes care of also starting a node backend in two separate processes and
 -- monitors both processes: if one terminates, then the other one is cancelled.
-execLaunchCommands :: Trace IO Text -> Maybe FilePath -> [Command] -> IO ()
-execLaunchCommands tracer stateDir commands = do
+execLaunch
+    :: Verbosity
+    -> Maybe FilePath
+    -> [Command]
+    -> IO ()
+execLaunch verbosity stateDir commands = do
     installSignalHandlers
-    maybe (pure ()) (setupStateDir tracer) stateDir
+    tracer <- initTracer (verbosityToMinSeverity verbosity) "launch"
+    maybe (pure ()) (setupStateDir $ logInfo tracer) stateDir
     logInfo tracer $ fmt $ nameF "launch" $ blockListF commands
     (ProcessHasExited pName code) <- launch commands
     logAlert tracer $ T.pack pName <> " exited with code " <> T.pack (show code)
     exitWith code
 
--- | Creates a command that can be used to start the wallet API server.
-commandWalletServe
-    :: String
-    -> Listen
-    -> Port "Node"
-    -> Maybe FilePath
-    -> String
-    -> Verbosity
-    -> Maybe (Hash "Genesis")
-    -> Command
-commandWalletServe
-        name listen backendPort stateDir network verbosity mGenesisHash =
-    Command name args (threadDelay oneSecond) Inherit
+{-------------------------------------------------------------------------------
+                              Options & Arguments
+-------------------------------------------------------------------------------}
+
+-- | --state=STRING
+addressStateOption :: Parser AddressState
+addressStateOption = optionT $ mempty
+    <> long "address"
+    <> metavar "STRING"
+    <> help "only addresses with the given state: either 'used' or 'unused'."
+
+-- | --database=FILEPATH
+databaseOption :: Parser FilePath
+databaseOption = optionT $ mempty
+    <> long "database"
+    <> metavar "FILEPATH"
+    <> help "use this file for storing wallet state. Run in-memory otherwise."
+
+-- | [--random-port|--port=INT]
+listenOption :: Parser Listen
+listenOption =
+    (ListenOnRandomPort <$ randomPortOption)
+    <|>
+    (ListenOnPort . getPort <$> portOption)
+
+-- | [--random-port]
+randomPortOption :: Parser Bool
+randomPortOption = flag' False $ mempty
+    <> long "random-port"
+    <> help "serve wallet API on any available port (conflicts with --port)"
+
+-- | [--node-port=INT], default: 8080
+nodePortOption :: Parser (Port "Node")
+nodePortOption = optionT $ mempty
+    <> long "node-port"
+    <> metavar "INT"
+    <> help "port used for communicating with the target node."
+    <> value (Port 8080)
+    <> showDefaultWith showT
+
+-- | --payment=PAYMENT
+paymentOption :: DecodeAddress t => Parser (AddressAmount t)
+paymentOption = optionT $ mempty
+    <> long "payment"
+    <> metavar "PAYMENT"
+    <> help
+        "address to send to and amount to send separated by @\
+        \, e.g. '<amount>@<address>'"
+
+-- | [--address-pool-gap=INT], default: 20
```

Comment:
What guarantee do we have that this value won't change? How can we keep the values in sync if the underlying default does change?

It turns out that this is defined within `Cardano.Wallet.Primitive.AddressDiscovery`.

Given that Haddock can make links to identifiers, perhaps we can write this instead:
```hs
 -- | [--address-pool-gap=INT], default: 'defaultAddressPoolGap'
```
(and similarly for other hard-coded constants).